core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java

```diff
@@ -87,14 +99,23 @@
     private TableId tableId;
     private ReadConsistency readConsistency = ReadConsistency.IMMEDIATE;
     private AccumuloClient _client;
+    private Collection<KeyExtent> extents;
 
     Builder(AccumuloClient client) {
       this._client = client;
     }
 
     @Override
     public TabletsMetadata build() {
-      Preconditions.checkState((level == null) != (table == null),
+      if (extents != null) {
+        // its expected that things are not set for scanning a single range, so check that. It also
+        // expected that checkConsistency is not set as this check can not be done for
+        // non-contiguous tablets.
```

Comment:
I didn't understand this check before, but now that you've written it out, I think a more concise way to put this might be something like:

```suggestion
        // setting multiple extents with forTablets(extents) is mutually exclusive with these single-tablet options
```